src/main/java/ome/services/blitz/repo/ManagedImportProcessI.java

```diff
@@ -346,7 +347,15 @@ public HandlePrx verifyUpload(List<String> hashes, Current __current)
                 final String clientHash = hashes.get(i);
                 String serverHash = "";
                 try {
-                    final RString result = (RString) iQuery.projection(hql, params, allGroupsContext).get(0).get(0);
+                    RString result;
+                    try {
+                        result = (RString) iQuery.projection(hql, params, groupContext).get(0).get(0);
+                    } catch (SecurityViolation sv) {
+                        /* The permissions context probably locks us to only certain groups. */
+                        log.debug("all-groups query for file checksum failed, retrying with current group context", sv);
```

Comment:
I definitely understand this as a workaround, it just seems that internal to the server we _should_ be able to force the system to give us what we want.

Reply:
Agreed, it's really just a case of how long it'd take to unpick the async import workflow enough to pick the right group, etc. A first step to a good solution would be to bring the group sudo status properly into the event context or the target group more clearly into the import context.

Reply:
Another option is to wire enough of the session provider in that we can determine what we need from that, but the code to both get it to this method then perform the test didn't seem worth the clutter; that could point to some other refactoring, like a security system method that can tell us a bit more.

Reply:
I like how this PR's approach here is probably fairly robust against changes in the details of how the group restriction permissions checking actually works.

Reply:
True, but since I don't know under what cases this is failing, I also don't know if it's sufficient. And, I would hate to see a general pattern for adoption be, "run every query twice".

Reply:
The only case in which I'm aware of this failing is when all-groups is not permitted because of a non-admin group sudo.

Another possible fix would be to rewrite the implementation of all-groups to detect such a situation and adjust the WHERE clause to something hairier.

Reply:
Sorry, I wasn't saying that your retrying here isn't ok as a workaround, I just don't want us to need to do that as a general rule to work around some deficiency in the allGroupContext.

Reply:
Definitely worth a broader refactoring when not trying to fix a painful bug at a busy time. :smiley: